app/Traits/ValidateContactRequests.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Traits;
+
+use Illuminate\Http\Request;
+use App\Models\Contact\Contact;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+
+trait ValidateContactRequests
```

Comment:
I don't understand why you use a trait to do this. You already have some kind of safe guard in `web.php` and `api.php`, and in `RouteServiceProvider`.

Reply:
TBH I was following how the contact pets resource was made. and this is pretty much exactly how it was done. so should I not follow the existing code as an example of how to implement a feature?

Reply:
to avoid duplicated code

Reply:
TBH I was following how the contact pets resource was made. and this is pretty much exactly how it was done. so should I not follow the existing code as an example of how to implement a feature?

Reply:
if policies are the way we should do it then why haven't we started?

Reply:
![image](https://user-images.githubusercontent.com/932221/50525258-05a40500-0ae3-11e9-9885-fd52a273a119.png)

Reply:
I think reafctoring how policies are handled was outside of the scope of this ticket